src/languageservice/utils/textBuffer.ts

```diff
@@ -0,0 +1,45 @@
+import { SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION } from 'constants';
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TextDocument } from 'vscode-languageserver';
+import { Range } from 'vscode-languageserver-types';
+
+export class TextBuffer {
+  constructor(private doc: TextDocument) {}
+
+  getLineCount(): number {
+    return this.doc.lineCount;
+  }
+
+  getLineLength(lineNumber: number): number {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const lineOffsets: number[] = (this.doc as any).getLineOffsets();
```

Comment:
out of curiosity because I'm not familiar with TypeScript. Why do you need to disable eslint rule? Is there another way to write it to avoid it?

Reply:
I cast to `any` to get access to 'internal' method.
Hm, it can be done without `any`